### ISSUE TITLE
weight parameter not working 

### ISSUE THREAD

**Issue by germayneng** [Microsoft/LightGBM#2023]
Good day,

I am interesting to use the weight param to set my weight for my objective function. I believe this weight give me the option to set individual weight to each row and this will in turn leads to higher weight == higher contribution in the terms of loss. 

Based on the documentation: 
![image](https://user-images.githubusercontent.com/22788747/53075633-a550a300-3528-11e9-95e5-8c1952e46d38.png)

I passed this as a dictionary params: (just a simple example). My weight column is the first column. 

```python 
params = {
        'nthread' : 6,
        'learning_rate': 0.01,
        'num_leaves':80,
        'max_depth': 7,
        'weight': 0}

```

and later used it in train. 

After training, when i check the feature importance and find that the column weight is used as a feature. Does this mean that the weight parameter is not taken into account? It will be good if you can help 

 Edit: is this weight_column suppose to be in .dataset and not in param dictionary?

**Comment by Laurae2** [comment]
@germayneng Use the Python API in Python for setting weights. You are using the documentation for CLI which behaves differently in wrappers.

See the following, the weights must be set separately: https://lightgbm.readthedocs.io/en/latest/Python-API.html#lightgbm.Dataset

**Comment by germayneng** [comment]
thanks @Laurae2 

One more thing, if i were to add to .Dataset, i assume that i only need to pass this weight array into the train dataset right? I should not need to pass it into the valid dataset?

**Comment by Laurae2** [comment]
weight should also affect metric. Passing weights to the validation set is possible.
